sql/log.cc

```diff
@@ -7970,6 +7972,20 @@ MYSQL_BIN_LOG::queue_for_group_commit(group_commit_entry *orig_entry)
     DBUG_ASSERT(entry != NULL);
     cur= entry->thd->wait_for_commit_ptr;
   }
+#ifdef WITH_WSREP
+  if (wsrep_is_active(entry->thd) &&
```

Comment:
MDEV-24915 changed the time window of when BF aborting happens. With current code, BF abort happens later and the victim may run much further. This group commit ordering issue happens in my testing for a victim which was BF aborted and started replaying. Here. the replayer and group commit leader were both concurrently accessing the THD::wsrep_client_state of the replayer. I have not encountered similar problem with earlier code versions.

Reply:
This fact that we allow victim to run much further is concerning because thread running victim will use time (read CPU time) and resources. All this is away from BF applier thread waiting for its turn. This could lead performance regression on multi-master loads.